XenServerHealthCheck/ServerListHelper.cs

```diff
@@ -267,5 +267,24 @@ public void UpdateProxy(string proxy)
                 log.Error("Error parsing 'ProxySetting' from XenCenter", e);
             }
         }
+
+        public void UpdateXenCenterMetadata(string message)
+        {
+            log.Info("Receive XenCenter metadata update message");
+
+            try
+            {
+                string[] metadata = message.Split(SEPARATOR);

```

Comment:
Good catch, I can't guarantee that the metadata won't contain the separator, especially as more items could be added in the future. Therefore I changed it to encode the metadata string before sending.